pkg/varnishcluster/controller/varnishcluster_statefulset.go

```diff
@@ -24,17 +24,259 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+func (r *ReconcileVarnishCluster) createVolumes(instance *vcapi.VarnishCluster) []v1.Volume {
+	varnishSecretName, varnishSecretKeyName := namesForInstanceSecret(instance)
+	volumes := []v1.Volume{
+		{
+			Name: vcapi.VarnishSharedVolume,
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: vcapi.VarnishSettingsVolume,
+			VolumeSource: v1.VolumeSource{
+				EmptyDir: &v1.EmptyDirVolumeSource{},
+			},
+		},
+		{
+			Name: vcapi.VarnishSecretVolume,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					Items: []v1.KeyToPath{
+						{
+							Key:  varnishSecretKeyName,
+							Path: "secret",
+							Mode: proto.Int32(0444), //octal mode read only
+						},
+					},
+					DefaultMode: proto.Int32(v1.SecretVolumeSourceDefaultMode),
+					SecretName:  varnishSecretName,
+				},
+			},
+		},
+	}
+	if instance.Spec.HaproxySidecar != nil && instance.Spec.HaproxySidecar.Enabled {
+		haproxyVolume := v1.Volume{
+			Name: vcapi.HaproxyConfigVolume,
+			VolumeSource: v1.VolumeSource{
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: instance.Spec.HaproxySidecar.ConfigMapName,
+					},
+				},
+			},
+		}
+		volumes = append(volumes, haproxyVolume)
+	}
+	return volumes
+}
+
+func (r *ReconcileVarnishCluster) createVarnishSharedVolumeMount(readOnly bool) v1.VolumeMount {
+	return v1.VolumeMount{
+		Name:      vcapi.VarnishSharedVolume,
+		MountPath: "/var/lib/varnish",
+		ReadOnly:  readOnly,
+	}
+}
+
+func (r *ReconcileVarnishCluster) createVarnishSettingsVolumeMount(readOnly bool) v1.VolumeMount {
+	return v1.VolumeMount{
+		Name:      vcapi.VarnishSettingsVolume,
+		MountPath: "/etc/varnish",
+		ReadOnly:  readOnly,
+	}
+}
+
+func (r *ReconcileVarnishCluster) createVarnishSecretVolumeMount() v1.VolumeMount {
+	return v1.VolumeMount{
+		Name:      vcapi.VarnishSecretVolume,
+		MountPath: "/etc/varnish-secret",
+		ReadOnly:  true,
+	}
+}
+
+func (r *ReconcileVarnishCluster) createVarnishContainer(instance *vcapi.VarnishCluster, varnishdArgs []string, varnishImage string) v1.Container {
+	//Varnish container
+	return v1.Container{
+		Name:  vcapi.VarnishContainerName,
+		Image: varnishImage,
+		Ports: []v1.ContainerPort{
+			{
+				Name:          vcapi.VarnishPortName,
+				ContainerPort: vcapi.VarnishPort,
+				Protocol:      v1.ProtocolTCP,
+			},
+		},
+		VolumeMounts: []v1.VolumeMount{
+			r.createVarnishSharedVolumeMount(false),
+			r.createVarnishSettingsVolumeMount(true),
+			r.createVarnishSecretVolumeMount(),
```

Comment:
those, and other similar methods doesn't use anything from the reconcile struct should be safe to have them as structs.